src/supervisor.adoc

```diff
@@ -2270,6 +2279,45 @@ Invalid PTEs using a bounded timer, or making address-translation caches
 coherent with store instructions that modify PTEs.
 ====
 
+[[sec:svukte]]
+== "Svukte" Extension for Address-Independent Latency of User-Mode Faults to Supervisor Addresses, Version 0.3
+
+The Svukte extension provides a means to make user-mode accesses to supervisor
+memory raise page faults in constant time, mitigating attacks that attempt to
+discover the supervisor software's address-space layout.
+
+If the Svukte extension is implemented, the `senvcfg`.UKTE field is writable.
+If the hypervisor extension is additionally implemented, the `hstatus`.HUKTE
+field is also writable.
+See <<sec:senvcfg>> and <<sec:hstatus>> for the definitions of those fields.
+
+The Svukte extension depends on Sv39.
+
+NOTE: Svukte is not defined for Sv32 because the small address space limits the
+available entropy, reducing the effectiveness of address-space layout
+randomization.
+If an Sv32 variant were to be defined, it would need to account for the fact
+that it is more common to reserve only the upper 1 GiB of the virtual-address
+space for the operating system, leaving the lower 3 GiB for user
+processes.
+
+When `senvcfg`.UKTE=1, an instruction fetch or explicit memory access whose
+effective privilege mode is U or VU is considered to be _Svukte-qualified_.
+For any Svukte-qualified memory access, virtual addresses {ge} 2^SXLEN-1^ are
+considered to be invalid; hence, an Svukte-qualified access to such an address
```

Comment:
> For any Svukte-qualified memory access, virtual addresses {ge} 2^SXLEN-1^ are
> considered to be invalid;

When CPU executes in VU-mode, do we also check 2^SXLEN-1 bit of virtual address (is invalid) in G-stage translation?

Reply:
I think the spec already says everything it needs to say. Svukte-qualified accesses are affected; the rest aren’t. Page-table walks aren’t Svukte-qualified.

Reply:
Thanks for confirmation!

Reply:
This doesn't explicitly say that the PTE.U bit is ignored, but it certainly implicitly does. I would make that explicit, and/or explain in section 10.31.1 that "The U bit indicates whether the page is accessible to user mode." [unless Svutke is enabled] or something

Reply:
The constraint on the implementation is much stricter than that. The implementation shouldn’t even perform the memory access corresponding to the PTE, and so it shouldn’t even be cognizant as to the value of PTE.U.

Don’t you think the next sentence in the spec (“The timing […] must be independent of the faulting VA”) gets this across? If not, how would you say it? Challenge: you cannot mention the PTE contents, given the machine’s behavior mustn’t depend on the PTE contents.